diffpy/snmf/subroutines.py

```diff
@@ -69,14 +69,26 @@ def get_stretched_component(stretching_factor, component, signal_length):
 
     Returns
     -------
-    1d array of floats
+    tuple of 1d array of floats
       The calculated component signal with stretching factors applied. Has length N, the length of the unstretched
-      component signal.
+      component signal. Also returns the gradient and hessian of the stretching transformation.
 
     """
+    delta = 1e-5
```

Comment:
I believe that np.gradient won't work in this case as np.gradient finds the gradient of an array based on the differences in the elements of the array whereas I am supposed to find the derivative of the stretching_function A(x) itself. I have found a well-maintained library numdifftools that allows me to do this in a much cleaner way.

Also, I'm not sure what you mean by "df/dstretch"

Reply:
Nice work, I think this looks better.  

df/dstretch was just trying to say what you said above in words (why you couldn't use numpy.gradient, because you are looking for the change of the function with the stretching factor (df/dstretch))